src/lib/icon/icon.component.ts

```diff
@@ -66,13 +67,22 @@ export class FaIconComponent implements OnChanges {
   @HostBinding('innerHTML')
   public renderedIconHTML: SafeHtml;
 
-  constructor(private sanitizer: DomSanitizer, private iconService: FaIconService) {}
+  constructor(
+      private sanitizer: DomSanitizer,
+      private iconService: FaIconService,
+      private renderer: Renderer2,
+      private elementRef: ElementRef,
+      @Host() @Optional() private stackParent: FaStackComponent
+      ) {}
 
   private params: IconParams;
   private iconSpec: IconLookup;
 
   ngOnChanges(changes: SimpleChanges) {
     if (changes) {
+      if (this.stackParent != null && this.size != null) {
```

Comment:
IMO it isn't a good solution to couple `FaIconComponent` to `FaStackComponent`. A cleaner solution would be to introduce a new directive to define/specify size of the stack item.

It may look something like:

```
<fa-stack size="2x">
    <fa-icon [icon]="faCircle" faStackSize="2x"></fa-icon>
    <fa-icon [icon]="faMobile" faStackSize="1x"></fa-icon>
    <!-- or -->
    <fa-icon [icon]="faCircle" faStackItem="2x"></fa-icon>
    <fa-icon [icon]="faMobile" faStackItem="1x"></fa-icon>
</fa-stack>
```

What do you think?

Also note that Font Awesome only has classes for `fa-stack-1x` and `fa-stack-2x`, so the type of the property must be `'1x' | '2x'`. `SizeProp` is for the icon, which supports much broader set of sizes.

Reply:
I've tried to implement the above suggestion and failed. Directives are change detected after their host component and we either have to render icon twice (first without fa-stack-* class and then once again to add it) or use dirty hacks to add this class from the directive after icon has been rendered (which is not trivial given that all icon markup is completely destroyed, when icon is re-rendered). So it doesn't seem to be possible to avoid making `fa-icon` aware about enclosing `fa-stack`.

I still would like to have a separate directive instead of changing behaviour of the `FaIcon.size` property as they serve different purposes and allow different set of values and I've updated the code accordingly.

I've also removed `FaStack.fixedWidth` input as stacks always have fixed width.